### PR TITLE
Read only buffers

### PR DESCRIPTION
**Pull request by bynect**
Initial implementation of read-only buffers (#30 ). Automatically sets read-only mode for manual files and can also detect if the file being open (in openFile and also in main) is read-only by using stat (I tested this on system file without root permissions and it worked).

### CODE REVIEW

**Comment by arthurbacci** [comment]
Conflict :pensive:

**Comment by arthurbacci** [comment]
I will merge it tomorrow, I can't now.

**Comment by bynect** [comment]
I fixed the conflicts, however while trying to change syntax config ted segfaulted. Seems like a bug was inserted in the latest commits, so I going to take a look at it

**Comment by bynect** [comment]
It was a fairly simple typo-bug, so I pushed the fix in this pr instead of creating another one.

File: src/keypress.c [hunk 1]
```diff
@@ -233,6 +249,5 @@ void process_keypress(int c) {
             lines[cy].ident = 0;
 
         syntaxHighlight();
-        syntaxHighlight();
```
Comment (bynect) [inline src/keypress.c:236]: Was this double call to syntaxHighlight intentional?

Comment (arthurbacci) [reply src/keypress.c:236]: No, maybe an automatic merge that didn't work correctly?

**Pull request merged**